LanguageFeatures/Inline-classes/static_analysis_member_invocation_A03_t01.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion If Dm is a getter declaration with return type R then the static
+/// type of the invocation is [T1/X1 .. Ts/Xs]R.
+///
+/// @description Checks static type of a getter of an `inline` class
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "../../Utils/static_type_helper.dart";
+
+inline class IC1 {
+  final num id;
+  IC1(this.id);
+
+  int get asInt => id as int;
+}
+
+inline class IC2<T extends num> {
+  final T id;
+  IC2(this.id);
+
+  double get asDouble => id as double;
```

Comment:
OK, that will work, but the following would perhaps be more natural, and it allows the test cases below to be written more freely:
```suggestion
  double get asDouble => id.toDouble();
```